workbench-core/accounts/src/services/projectService.ts

```diff
@@ -80,4 +71,211 @@ export default class ProjectService {
 
     return Promise.resolve({ data: projectsResponse.Items as unknown as Project[] });
   }
+
+  /**
+   * Creates a new project
+   *
+   * @param params - the required fields to create a new project
+   * @param user - authenticated user creating the project
+   * @returns Project object of new project
+   */
+  public async createProject(params: CreateProjectReqeust, user: AuthenticatedUser): Promise<Project> {
+    // Verify caller is an IT Admin--TODO implement after dynamic AuthZ
+    // const userGroupsForCurrentUser: string[] = await this._dynamicAuthorizationService.getUserGroups(user.id);
+    // if(userGroupsForCurrentUser.length !== 1 || userGroupsForCurrentUser[0] !== 'ITAdmin') {
+    //   throw Boom.forbidden('Only IT Admins are allowed to create new Projects.');
+    // }
+
+    // Verify project name is unique and cost center exists
+    const [isProjectNameInUse, costCenter] = await Promise.all([
+      this._isProjectNameInUse(params.name),
+      this._getCostCenter(params.costCenterId)
+    ]);
+    if (isProjectNameInUse) {
+      throw Boom.badRequest('Project Name is in use by a non deleted project. Please use another name.');
```

Comment:
Good idea. I was torn on this because originally I was checking existence on both and not saving objects but then I elected to use the cost center service getCostCenter method. I will change so it is this way.